File: CS/ImageMemoEdit/CustomGridControl/SmartIconSelectionEventArgs.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System.ComponentModel;
using EditorDescendant;
using System.Drawing;

namespace ImageTextEdit
{
    public class SmartIconSelectionEventArgs : IconSelectionEventArgs
    {
        bool isActiveEditor;
        GridCellInfo cellInfo;
        public SmartIconSelectionEventArgs(Image image, GridCellInfo cellInfo, bool isActiveEditor)
            : base(image)
        {
            this.cellInfo = cellInfo;
            this.isActiveEditor = isActiveEditor;
        }
        public GridCellInfo CellInfo { get { return cellInfo; } }
        public bool IsActiveEditor { get { return isActiveEditor; } }
    }
}

File: VB/ImageMemoEdit/CustomGridControl/SmartIconSelectionEventArgs.vb
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System.ComponentModel
Imports EditorDescendant
Imports System.Drawing

Namespace ImageTextEdit

    Public Class SmartIconSelectionEventArgs
        Inherits IconSelectionEventArgs

        Private isActiveEditorField As Boolean

        Private cellInfoField As GridCellInfo

        Public Sub New(ByVal image As Image, ByVal cellInfo As GridCellInfo, ByVal isActiveEditor As Boolean)
            MyBase.New(image)
            cellInfoField = cellInfo
            isActiveEditorField = isActiveEditor
        End Sub

        Public ReadOnly Property CellInfo As GridCellInfo
            Get
                Return cellInfoField
            End Get
        End Property

        Public ReadOnly Property IsActiveEditor As Boolean
            Get
                Return isActiveEditorField
            End Get
        End Property
    End Class
End Namespace
